pallets/providers/src/utils.rs

```diff
@@ -662,16 +664,7 @@ where
             Error::<T>::NewCapacityLessThanUsedStorage
         );
 
-        // Calculate how much deposit will the signer have to pay to register with this amount of data
-        let capacity_over_minimum = new_capacity
-            .checked_sub(&T::SpMinCapacity::get())
-            .ok_or(Error::<T>::StorageTooLow)?;
-        let deposit_for_capacity_over_minimum = T::DepositPerData::get()
-            .checked_mul(&capacity_over_minimum.into())
-            .ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
-        let new_deposit = T::SpMinDeposit::get()
-            .checked_add(&deposit_for_capacity_over_minimum)
-            .ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
+        let new_deposit = Self::compute_deposit_needed_for_capacity(new_capacity)?;
```

Comment:
There are 3 other places where this calculation is done. We'd want to unify that.